### ISSUE TITLE
Possible problem when inputs are duplicated

### ISSUE THREAD

**Issue by jkeifer** [cirrus-geo/cirrus-geo#84]
Saw a traceback from the cirrus `process` function calling `Catalogs.process()`:

```
Traceback (most recent call last):
File "/var/task/cirruslib/statedb.py", line 80, in get_dbitems
'Keys': [self.catid_to_key(id) for id in catids]
File "/var/runtime/botocore/client.py", line 386, in _api_call
return self._make_api_call(operation_name, kwargs)
File "/var/runtime/botocore/client.py", line 705, in _make_api_call
raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the BatchGetItem operation: Provided list of item keys contains duplicates
```

Seems like a bug where duplication of an input item combined with lambda message batching can result in the process lambda trying to process all the duplicated messages at the same time, and then the state DB query fails due to duplicated item IDs.

We likely should not allow duplicate items to be added to the `Catalogs` object, but how best to handle that could be tricky. On one hand we'd want to delete the duplicate message, but on the other how do we know they are identical, and if not, which is the "better" one? Perhaps we can treat it as first always wins, because if message batching didn't have the duplicates processed via a single execution then that would be the effective result with the statedb lookup anyway.

Based on the above complication, I am making this issue here rather than in cirrus-lib, even though the latter will also require modification. Once we determine what that is we can make a more specific ticket on that side to track that work.

**Comment by jkeifer** [comment]
After further consideration, I still feel that first processed wins, and any additional payloads for the same input items simply get ignored.

**Comment by matthewhanson** [comment]
Not sure I quite understand the issue - the error is happening in `get_dbitems` because, presumably, the array passed to `get_batch_item` contains duplicate IDs.

So can't you just create the array first and take the set to ensure it's only unique IDs?

```
self.payload_id_to_key(id) for id in set(payload_ids)
```

or, is it assuming a fixed size array in the response that corresponds to IDs that were passed into the function?

**Comment by jkeifer** [comment]
@matthewhanson The problem is not deduplicating the IDs in the dynamo query; you are right, that part is easy. The question is how we are going to handle multiple payloads for the same input IDs. Do we just assume the first payload we receive for a given set of inputs is the one we process, and we drop any additional ones? I think that is all we can do, but I wanted to bounce that off you to make sure I wasn't missing something.

**Comment by jkeifer** [comment]
Fixed by cirrus-lib release 0.6.2.
